Orm/Xtensive.Orm/Orm/Linq/Translator.Queryable.cs

```diff
@@ -28,9 +28,35 @@ internal sealed partial class Translator : QueryableVisitor
   {
     private static readonly Type IEnumerableOfKeyType = typeof(IEnumerable<Key>);
 
-    public TranslatorState state;
+    internal TranslatorState state { get; private set; } = TranslatorState.InitState;
     private readonly TranslatorContext context;
 
+    internal void RestoreState(in TranslatorState previousState) =>
+      state = previousState;
+
+    public TranslatorState.TranslatorScope CreateScope(in TranslatorState newState)
+    {
+      var scope = new TranslatorState.TranslatorScope(this);
+      state = newState;
+      return scope;
+    }
+
+    public TranslatorState.TranslatorScope CreateLambdaScope(LambdaExpression le, bool allowCalculableColumnCombine)
+    {
+      var newOuterParameters = new ParameterExpression[state.OuterParameters.Length + state.Parameters.Length];
+      state.OuterParameters.CopyTo(newOuterParameters, 0);
+      state.Parameters.CopyTo(newOuterParameters, state.OuterParameters.Length);
+      return CreateScope(new TranslatorState(state) {
+        OuterParameters = newOuterParameters,
+        Parameters = le.Parameters.ToArray(le.Parameters.Count),
+        CurrentLambda = le,
+        AllowCalculableColumnCombine = allowCalculableColumnCombine
+      });
+    }
+
+    private void ModifyStateAllowCalculableColumnCombine(bool b) =>
+      state = new TranslatorState(state) { AllowCalculableColumnCombine = b };
+
```

Comment:
Do we need these utils-like methods at the very top? Are they that important for a reader?

Reply:
Are you sure? Maybe you forget to push this change?

Reply:
> Are you sure? Maybe you forget to push this change?

https://github.com/DataObjects-NET/dataobjects-net/pull/170/files#diff-4ebdc09d5003e6d010bb0aab7fda2db5144d843d0cafa127bd2c88f79a629491R1807

Reply:
moved all of them and converted to private